### PR TITLE
Disallow template literals on colors and modifiers in the TypeScript types

### LINKED ISSUE #341: Template literals are unsupported for nested calls
[Issuehunt badges](https://issuehunt.io/r/chalk/chalk/issues/341)

The following is unsupported despite it being allowed by the type system:
```ts
chalk.bold`Hello, {cyan.inverse ${name}!} This is a test. {green ${exclamation}!}`;
```
Produces:
```
Hello, {cyan.inverse ,!} This is a test. {green ,!} Sindre Neat
```
Instead of:
```
Hello, Sindre! This is a test. Neat!
```

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Submitted pull Requests
- [#392 Support for template literals for nested calls](https://issuehunt.io/r/chalk/chalk/pull/392)
- [#398 Support nested calls](https://issuehunt.io/r/chalk/chalk/pull/398)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/chalk/chalk/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by sexwithsatan**
Fixes #341

### CODE REVIEW

**Comment by sexwithsatan** [comment]
Any feedback on this? PR was submitted over 3 weeks ago

**Comment by sindresorhus** [comment]
@ExE-Boss Can you help review?

**Comment by ExE-Boss** [comment]
This&nbsp;won’t fix&nbsp;<https://github.com/chalk/chalk/issues/341>, this&nbsp;just&nbsp;makes the&nbsp;**TypeScript** type&nbsp;definition mark&nbsp;attempting&nbsp;it as&nbsp;an&nbsp;error until&nbsp;<https://github.com/chalk/chalk/issues/341>&nbsp;is&nbsp;implemented.

File: index.d.ts [hunk 1]
```diff
@@ -398,7 +411,7 @@ Call the last one as a method with a string argument.
 Order doesn't matter, and later styles take precedent in case of a conflict.
 This simply means that `chalk.red.yellow.green` is equivalent to `chalk.green`.
 */
-declare const chalk: chalk.Chalk & chalk.ChalkFunction & {
+declare const chalk: chalk.Chalk & chalk.ChalkFunction & chalk.ChalkTemplateFunction & {
```
Comment (sindresorhus) [inline index.d.ts]: Isn't `& chalk.ChalkFunction & chalk.ChalkTemplateFunction` moot here as `chalk.Chalk` already extend those?

**Pull request closed without merging**